c7n/resources/s3.py

```diff
@@ -3114,6 +3114,10 @@ def resolve_keys(self, buckets):
                 self.arns[r] = client.describe_key(
                     KeyId=self.data.get('key')
                 ).get('KeyMetadata').get('Arn')
+                if self.data.get('key').startswith('alias/aws'):
```

Comment:
is there a reason we only want to do this with aws aliased keys vs also doing it with customer keys? maybe i'm missing it but KeyId on describe_key above supports arn or alias afaicr. is the intent here to preserve an alias when putting it ? or to support the alias on describe, the integration with kms on alias support is a bit service specific, i've seen some take alias but return ids..

Reply:
i got context from the issue this in support of.